bindings/dotnetcore/src/dotnetcore.cpp

```diff
@@ -34,13 +34,13 @@ typedef void(*PGatewayDestroyDelegate)(unsigned int moduleIdManaged);
 
 typedef void(*PGatewayStartDelegate)(unsigned int moduleIdManaged);
 
-static PGatewayCreateDelegate GatewayCreateDelegate;
```

Comment:
I don't know where this is called, so I'm assuming someone could call it with different delegates. In WIndows, each DLL will have a separate instance of these globals, but Linux might not.  If the user can set these delegates, then that might have interesting side-effects.

The other way to unit test is to also create a "GetBindingDelegates" and test them together.

---
In reply to: [104541524](https://github.com/Azure/azure-iot-gateway-sdk/pull/149#discussion_r104541524) [](ancestors = 104541524)